stringfog-gradle-plugin/src/main/java/com/github/megatronking/stringfog/plugin/StringFogInstrumentationParams.kt

```diff
@@ -0,0 +1,53 @@
+package com.github.megatronking.stringfog.plugin
+
+import com.android.build.api.instrumentation.InstrumentationParameters
+import com.github.megatronking.stringfog.StringFogWrapper
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import java.lang.ref.WeakReference
+import java.util.WeakHashMap
+
+abstract class StringFogInstrumentationParams : InstrumentationParameters {
+    @get:Input
+    abstract val applicationId: Property<String>
+
+    @get:Input
+    abstract val className: Property<String>
+}
+
+private class NonSerializableParams(
+    val logs: MutableList<String>,
+    val implementation: StringFogWrapper
+)
+
+private val extensionForApplicationId = mutableMapOf<String, WeakReference<StringFogExtension>>()
+private val extensionNonSerializableParams = WeakHashMap<StringFogExtension, NonSerializableParams>()
+
+internal val StringFogInstrumentationParams.extension
+    get() = extensionForApplicationId[applicationId.get()]?.get()
+        ?: throw IllegalStateException("Extension has not been registered with setParameters")
+
+private val StringFogInstrumentationParams.nonSerializableParameters
+    get() = extension.let { extensionNonSerializableParams[it] }
+        ?: throw IllegalStateException("runtimeParameters have not been registered with setParameters")
+
+internal val StringFogInstrumentationParams.logs get() = nonSerializableParameters.logs
+
+internal val StringFogInstrumentationParams.implementation get() = nonSerializableParameters.implementation
+
+internal fun StringFogInstrumentationParams.setParameters(
+    applicationId: String,
+    extension: StringFogExtension,
+    logs: MutableList<String>,
+    className: String
+) {
+    this.applicationId.set(applicationId)
+    this.className.set(className)
+    extensionForApplicationId[applicationId] = WeakReference(extension)
+    extensionNonSerializableParams[extension] = NonSerializableParams(
```

Comment:
Can we use `applicationId` as the map key?

Reply:
It's probably safe to use `applicationId` as map key and skip the `WeakHashMap` and `WeakReference`.

I just used that extra step of indirection so that the `NonSerializableParams` and anything in it could be garbage collected as soon as the extension is GC:ed. 

It's just a general habit I've picked up of not storing anything I'm not confident in the memory characteristics of in singletons, but it is almost certainly overkill in this case.

Reply:
Thanks, I will merge this PR.